sys_util/src/lib.rs

```diff
@@ -18,9 +18,46 @@ mod signal;
 mod struct_util;
 mod terminal;
 
-pub use errno::{errno_result, Error, Result};
+pub use errno::Error;
 pub use eventfd::*;
 pub use ioctl::*;
 pub use signal::*;
-pub use struct_util::*;
+pub use struct_util::{read_struct, read_struct_slice};
 pub use terminal::*;
+
+#[doc(hidden)]
+pub trait IsMinusOne {
```

Comment:
Is it really necessary to implement this trait types other than `c_int`? It looks like the only type we're ever checking against -1 is `c_int`.

Reply:
Not really, maybe it is a bit overkill. I can remove the trait and hardcode `c_int` if you guys want so.